### PR TITLE
update datamodels documentation

### PR DESCRIPTION
**Pull request by jdavies-st**
First attempt at updating datamodels documentation.  More to be done.  Current updates:

- added all the new datamodel types since Build 6
- examples use python 3 print() syntax now

### CODE REVIEW

File: docs/datamodels/source/devel/attributes.rst [hunk 1]
```diff
@@ -17,34 +17,41 @@ List of current models
...
+steps.
+
+For uncalibrated data `_uncal.fits`.  Getting the number of integrations
+and the number of groups from the first and second axes assumes that the
+input data array is 4-D data.  Pixel coordinates in the data extensions are
+1-indexed as in FORTRAN and FITS headers, not 0-indexed as in Python.
+
+    - ``input_model.data.shape[0]``: number of integrations
+    - ``input_model.data.shape[1]``: number of groups
+    - ``input_model.meta.exposure.nframes``: number of frames
```
Comment (hbushouse) [inline docs/datamodels/source/devel/attributes.rst]: Might want to label this as "number of frames per group"

File: docs/datamodels/source/devel/attributes.rst [hunk 2]
```diff
@@ -17,34 +17,41 @@ List of current models
...
+    - ``input_model.meta.exposure.groupgap``: group gap
```
Comment (hbushouse) [inline docs/datamodels/source/devel/attributes.rst]: "number of frames dropped between groups"

File: docs/datamodels/source/devel/attributes.rst [hunk 3]
```diff
@@ -61,26 +68,29 @@ is stored in a separate image extension).  Full-frame data for all other
...
+DataModel Base Class
+--------------------
 
 .. autoclass:: jwst.datamodels.DataModel
    :members:
 
-Concrete model classes
-''''''''''''''''''''''
+Specific Model Classes
+----------------------
```
Comment (hbushouse) [inline docs/datamodels/source/devel/attributes.rst]: How is this list of model classes different than the one shown at the top of the doc ("current models")?

Comment (jdavies-st) [inline docs/datamodels/source/devel/attributes.rst]: It's the same list, but it uses the Sphinx `automodule` to generate the docstring documentation for each DataModel class.  The list at the top merely serves as a link to each chunk of documentation at the bottom.

**Pull request merged**